evennia/contrib/actions/README.md

```diff
@@ -0,0 +1,372 @@
+ 
+Finally, the evennia.contrib.actions module supplies the __Action__ class, which all the actions you design will inherit. A thorough description of this class can be found further down this document in its own [section](#actions). This is how you import the Action class:
+
+> from evennia.contrib.actions import Action
+
+
+## The ActionCharacter and ActionRoom classes
+
+The full functionality of the ActionCharacter and ActionRoom classes can be accessed through a single attribute of the class, "actions". This attribute has several properties that you might sometimes want to change, as well as methods you will want to call or overload and a few things that you should set in the at_object_creation method as described in the section above.
+
+ActionCharacter objects have the attribute "actions.active". Switching this from True to False prohibits the character from performing any actions that do not have their "passive" value set to True, and cancels all prohibited actions that are being carried out by the character at present. Switching this from False to True allows the character to perform actions again and makes them begin the first of their queued actions immediately, if any such actions exist. You can use this attribute to simulate effects such as paralysis and death.
+
+The ActionCharacter's "actions.done()" method may be called to have the character end their turn. This is automatically implemented in the "done" command (see below).
+
+The "actions.stop(ongoing=True, queued=False)" method stops all of the character's ongoing and/or queued actions, based on the parameters given to it.
+The "stop" command automatically implements this method (see below).
+
+Finally, the "actions.unpuppet" command stops all of the character's cancellable actions and turns off their alarm (you should make sure that non-cancellable actions never have their "invokes_tb" item set to True, because this would prevent the alarm from being turned off and lock the room in turn-based mode). You should probably call this in your character's . 
+
+ActionRoom objects have the method "actions.display(viewed_object, message, target=None, data="", default=False)", which will show everyone in the room a processed version of the message string in the arguments list according to the room's view function (if you have set it) or in a default way if no view function exists. The target should be an object, while the data should be a string or an object. If default is set to True, and the viewed object is invisible to a given viewer, the name of the viewed object will be replaced (with "Someone" if the object is a character or "Something" otherwise) in the version of the message given to that viewer. If default is setto False, and the viewed object is invisible to the viewer, the message will not be shown to the viewer at all. Thus, the display method can be used to implement some features of stealth. 
```

Comment:
Markdown formatting point a function example like this should be put in `backtick` markers rather than quotes.